File: pbmx_ffi/java/io/rmf/pbmx/ShuffleProof.java
package io.rmf.pbmx;

import io.rmf.pbmx.ffi.LibPbmx;
import com.sun.jna.Pointer;

public final class ShuffleProof {
    ShuffleProof(Pointer handle) {
        this.handle = handle;
    }

    @Override
    protected void finalize() {
        LibPbmx.INSTANCE.pbmx_delete_shuffle_proof(this.handle);
    }

    Pointer handle;
}

File: pbmx_droid/app/src/main/java/io/rmf/pbmx/ShuffleProof.kt
package io.rmf.pbmx

import com.sun.jna.Pointer
import io.rmf.pbmx.ffi.FFI

class ShuffleProof internal constructor(internal var handle: Pointer) {

    protected fun finalize() {
        FFI.pbmx_delete_shuffle_proof(this.handle)
    }
}
